ingest-app/src/cmr/ingest/data/granule_bulk_update.clj

```diff
@@ -0,0 +1,253 @@
+             (update :request-json-body util/gzip-blob->string)
+             (update :created-at (partial oracle/oracle-timestamp->str-time conn)))))
+
+ (get-bulk-update-task-granule-status
+   [db task-id]
+   ;; Get statuses for all granules by task id
+   (map util/map-keys->kebab-case
+        (su/query db (su/build (su/select [:granule-ur :status :status-message]
+                                          (su/from "bulk_update_gran_status")
+                                          (su/where `(= :task-id ~task-id)))))))
+
+ (get-bulk-update-granule-status
+   [db task-id granule-ur]
+   ;; Get the status for a particular granule
+   (su/find-one db (su/select [:status :status-message]
+                              (su/from "granule_bulk_update_tasks")
+                              (su/where `(and (= :task-id ~task-id)
+                                              (= :granule-ur ~granule-ur))))))
+
+ (create-and-save-bulk-update-status
+   [db provider-id json-body granule-urs]
+   ;; In a transaction, add one row to the task status table and for each concept
+   (j/with-db-transaction
+     [conn db]
+     (let [task-id (:nextval (first (su/query db ["SELECT task_id_seq.NEXTVAL FROM DUAL"])))
+           statement (str "INSERT INTO bulk_update_task_status "
+                          "(task_id, provider_id, name, request_json_body, status)"
+                          "VALUES (?, ?, ?, ?, ?)")
+           name (get (json/parse-string json-body) "name" task-id)
+           values [task-id provider-id name (util/string->gzip-bytes json-body) "IN_PROGRESS"]]
+       (j/db-do-prepared db statement values)
+       ;; Write a row to granule status for each concept id
+       (apply j/insert! conn
+              "granule_bulk_update_tasks"
+              ["task_id" "granule-ur" "status"]
+              ;; set :transaction? false since we are already inside a transaction
+              (concat (map #(vector task-id % "PENDING") granule-urs) [:transaction? false]))
+       task-id)))
+
+ (update-bulk-update-task-status
+   [db task-id status status-message]
+   (try
+     (let [statement (str "UPDATE bulk_update_gran_status "
+                          "SET status = ?, status_message = ?"
+                          "WHERE task_id = ?")]
+       (j/db-do-prepared db statement [status status-message task-id]))
+     (catch Exception e
+       (errors/throw-service-error :invalid-data
+                                   [(str "Error creating updating bulk update task status "
+                                         (.getMessage e))]))))
+
+ (update-bulk-update-granule-status
+   [db task-id granule-ur status status-message]
+   (try
+     (j/with-db-transaction
+      [conn db]
+      (let [statement (str "UPDATE granule_bulk_update_tasks "
+                           "SET status = ?, status_message = ?"
+                           "WHERE task_id = ? AND granule_ur = ?")
+            status-message (util/trunc status-message 4000)]
```

Comment:
what is the significance of 4000?

Reply:
This is just what collection bulk update used, but I agree that this magic number should be changed

Reply:
it should match what is defined in db schema.